experimental/tinystr_neo/src/macros.rs

```diff
@@ -4,14 +4,12 @@
 
 #[macro_export]
 macro_rules! tinystr {
-    ($n:literal, $s:literal) => {
-        ({
-            // Force it into a const context; otherwise it may get evaluated at runtime instead.
-            const TINYSTR_MACRO_CONST: Result<$crate::TinyAsciiStr<$n>, $crate::TinyStrError> = $crate::TinyAsciiStr::from_str($s);
-            TINYSTR_MACRO_CONST
-        // Note: Rust 1.57 allows panics in consts, until we update to that version we have to panic outside of the const and hope it optimizes away
-        }).unwrap()
-    }
+    ($n:literal, $s:literal) => {{
+        // Force it into a const context; otherwise it may get evaluated at runtime instead.
+        const TINYSTR_MACRO_CONST: $crate::TinyAsciiStr<$n> =
+            $crate::TinyAsciiStr::from_str_panicky($s);
```

Comment:
Suggestion: does it work if you do this?

```suggestion
        const TINYSTR_MACRO_CONST: $crate::TinyAsciiStr<$n> =
            $crate::TinyAsciiStr::from_str($s).unwrap();
```

or perhaps

```suggestion
        const TINYSTR_MACRO_CONST: $crate::TinyAsciiStr<$n> =
            $crate::TinyAsciiStr::from_str($s).ok().expect("Invalid tinystr");
```

Reply:
(with the `unwrap` inside the const context)

Reply:
No, unwrap isn't const (because it calls formatting infra)

Reply:
Not even `Option::unwrap`?  oh well... how about

```rust
const TINYSTR_MACRO_CONST: $crate::TinyAsciiStr<$n> =
            match $crate::TinyAsciiStr::from_str($s) {
    Ok(s) => s,
    Err(_) => panic!("{}", $s)  // Invalid tinystr
}
```

Or, keep it the way it is.  It's just that it might be nice to not have a panicky library function.

Reply:
Yeah I didn't like that function either, moved